AdvancedTradeSkillWindow2/AdvancedTradeSkillWindow2.lua

```diff
@@ -4151,34 +4152,43 @@ local function GetReagents(Name, Amount, Link, Texture, Table)
 	return Table
 end
 
+--- This entire function has been rewritten by Ayri in August 2024 as part of bugfix ---
 local function GetSayTooltipString()
 	if ChatFrameEditBox:IsVisible() or WIM_EditBoxInFocus then
 		local ChatType, ChatNumber, ChatName, ChatID
 		local SayTarget = ""
-		
+
 		if WIM_EditBoxInFocus ~= nil then
-			ChatType 		= "WHISPER"
-			ChatNumber 	= WIM_EditBoxInFocus:GetParent().theUser
-			
-			SayTarget = ATSW_TOOLTIP_TO .. WIM_UserWithClassColor(ChatNumber)
+			ChatType = "WHISPER"
+			ChatNumber = WIM_EditBoxInFocus:GetParent().theUser
+
+			SayTarget= ATSW_TOOLTIP_TO .. WIM_UserWithClassColor(ChatNumber)
 		else
 			ChatType, ChatNumber = ChatFrameEditBox.chatType
-			
-			if 			ChatType == "WHISPER" then ChatNumber = ChatFrameEditBox.tellTarget
-			elseif	ChatType == "CHANNEL" then ChatNumber = ChatFrameEditBox.channelTarget end
-			
-			if			ChatType ~= "SAY"	   	 then
-				ChatID, ChatName = GetChannelName(ChatNumber)
-				
-				SayTarget = ATSW_TOOLTIP_TO .. "[" .. ChatID .. ". " .. ChatName .."]"
-			end
+
+		if ChatType == "WHISPER" then
+				ChatNumber = ChatFrameEditBox.tellTarget
+		elseif ChatType == "CHANNEL" then
+				ChatNumber = ChatFrameEditBox.channelTarget
```

Comment:
Add this after this line:
```
					ChatID, ChatName = GetChannelName(ChatNumber)
				
					SayTarget = ATSW_TOOLTIP_TO .. "[" .. ChatID .. ". " .. ChatName .."]"
```

Reply:
--- This entire function has been rewritten by Ayri in August 2024 as part of bugfix ---
local function GetSayTooltipString()
	if ChatFrameEditBox:IsVisible() or WIM_EditBoxInFocus then
		local ChatType, ChatNumber, ChatName, ChatID
		local SayTarget = ""

		if WIM_EditBoxInFocus ~= nil then
			ChatType = "WHISPER"
			ChatNumber = WIM_EditBoxInFocus:GetParent().theUser

			SayTarget= ATSW_TOOLTIP_TO .. WIM_UserWithClassColor(ChatNumber)
		else
			ChatType, ChatNumber = ChatFrameEditBox.chatType

		if ChatType == "WHISPER" then
				ChatNumber = ChatFrameEditBox.tellTarget
		elseif ChatType == "CHANNEL" then
				ChatNumber = ChatFrameEditBox.channelTarget
		elseif ChatType == "YELL" then
				SayTarget = ATSW_TOOLTIP_TO .. "[YELL]"
		elseif ChatType == "PARTY" then
				SayTarget = ATSW_TOOLTIP_TO .. "[PARTY]"
		elseif ChatType == "GUILD" then
				SayTarget = ATSW_TOOLTIP_TO .. "[GUILD]"
		elseif ChatType == "RAID" then
				SayTarget = ATSW_TOOLTIP_TO .. "[RAID]"

			SayTarget = ATSW_TOOLTIP_TO .. "[" .. ChatID .. ". " .. ChatName .. "]"
		end
	end

		return ATSW_TOOLTIP_SAYREAGENTS .. SayTarget, ChatType, ChatNumber
	else
		return ""
	end
end
--- End of rewrite ---

this is the entire block of code; it is a bit janky (using multiple elseif statements) because LUA doesn't really have a 'switch' functionality

Reply:
I don't believe adding this code will help.  I've tested the changes in-game using my existing code changes to this file and it works as expected in every chat channel including whispers and otherwise.

Reply:
I don't require or ask for any credit.  If you want to manually edit this version of the file in your repository and commit it as a bugfix you are welcome to do so.

I believed you had 'quit' WoW (per your ReadMe) and weren't returning to the game.  If you would like to take over maintenance of the addon please let me know.

Reply:
I'll freely admit I'm still new to github and how it works so my commit history is a bit messy.